src/workerd/api/http.c++

```diff
@@ -1603,8 +1597,11 @@ jsg::Promise<jsg::Ref<Response>> fetchImplNoOutputLock(
       };
 
       // TODO(someday): Allow deferred proxying for bidirectional streaming.
-      ioContext.addWaitUntil(handleCancelablePump(js, ioContext, signal, jsBody, kj::mv(stream)));
-
+      ioContext.addWaitUntil(handleCancelablePump(
+          AbortSignal::maybeCancelWrap(signal,
+              ioContext.waitForDeferredProxy(
+                  jsBody->pumpTo(js, kj::mv(stream), true))))
+                      .attach(jsBody.addRef()));
```

Comment:
Huh, it looks like the old code didn't actually do anything to attach the `jsBody` (it only passes a reference into the coroutine lambda, which is immediately invalidated on `co_await`). But I can't quite tell if that was a bug, which this is actually fixing.

Minor, but you could save an allocation by passing the `jsBody.addRef()` into the coroutine to be stored on its frame, instead of as an attachment.

Reply:
Yep, there's certainly a possible bug there, Difficult to say for sure but happy that I caught it